common/services/file-extension.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Get the file extension from a file buffer. Calculates the image format by reading the first 4 bytes of the image (header)
+ * Supported types: jpeg, jpg, png, svg
+ * Additionally, this function allows checking gif images.
+ * @param buffer file buffer
+ * @returns the file extension. Example: jpg, png, svg. it Returns unknown if it can not find the extension.
+*/
+export function getFileExtensionFromBuffer(buffer: Buffer): string {
+	const imageFormat = buffer.toString('hex').substring(0, 4);
+	switch (imageFormat) {
+		case '4749':
+			return 'gif';
+		case 'ffd8':
+			return 'jpg'; // Also jpeg
+		case '8950':
+			return 'png';
+		case '3c3f':
```

Comment:
I did some research and looks like `3c73` is also a valid file signature for SVG files.

Lots of examples [here](https://www.w3schools.com/graphics/svg_examples.asp). Copy-paste and remove the HTML tags:

```xml
<svg height="400" width="450">
<path id="lineAB" d="M 100 350 l 150 -300" stroke="red" stroke-width="3" fill="none" />
  <path id="lineBC" d="M 250 50 l 150 300" stroke="red" stroke-width="3" fill="none" />
  <path d="M 175 200 l 150 0" stroke="green" stroke-width="3" fill="none" />
  <path d="M 100 350 q 150 -300 300 0" stroke="blue" stroke-width="5" fill="none" />
  <!-- Mark relevant points -->
  <g stroke="black" stroke-width="3" fill="black">
    <circle id="pointA" cx="100" cy="350" r="3" />
    <circle id="pointB" cx="250" cy="50" r="3" />
    <circle id="pointC" cx="400" cy="350" r="3" />
  </g>
  <!-- Label the points -->
  <g font-size="30" font-family="sans-serif" fill="black" stroke="none" text-anchor="middle">
    <text x="100" y="350" dx="-30">A</text>
    <text x="250" y="50" dy="-10">B</text>
    <text x="400" y="350" dx="30">C</text>
  </g>
  Sorry, your browser does not support inline SVG.
</svg>
```

![image](https://user-images.githubusercontent.com/15186973/198055884-973bc0bb-9484-41ab-8ed4-831c85aaa4c3.png)